db-tests/src/test/java/gov/nasa/jpl/aerie/database/ExternalEventTests.java

```diff
@@ -0,0 +1,2168 @@
+      // create the source type
+      statement.executeUpdate(
+          // language=sql
+          """
+          INSERT INTO
+            merlin.external_source_type
+          VALUES ('%s');
+          """.formatted(source_type_name)
+      );
+    }
+  }
+
+  protected void insertDerivationGroup(String derivation_group_name, String source_type_name) throws SQLException {
+    try(final var statement = connection.createStatement()) {
+      // create the derivation group
+      statement.executeUpdate(
+          // language=sql
+          """
+          INSERT INTO
+            merlin.derivation_group
+          VALUES ('%s', '%s');
+          """.formatted(derivation_group_name, source_type_name)
+      );
+    }
+  }
+
+  protected void insertExternalSource(ExternalSource externalSource) throws SQLException {
+    try(final var statement = connection.createStatement()) {
+      System.out.println("STARTING " + externalSource);
+      // create the source
+      statement.executeUpdate(
+          // language=sql
+          """
+          INSERT INTO
+            merlin.external_source
+          VALUES ('%s', '%s', '%s', '%s', '%s', '%s', '%s');
+          """.formatted(
+              externalSource.key,
+              externalSource.source_type_name,
+              externalSource.derivation_group_name,
+              externalSource.valid_at,
+              externalSource.start_time,
+              externalSource.end_time,
+              externalSource.created_at
+          )
+      );
+      System.out.println("FINISHED " + externalSource);
+    }
+  }
+
+  protected void insertExternalEvent(ExternalEvent externalEvent) throws SQLException {
+    try(final var statement = connection.createStatement()) {
+      // create the event
+      statement.executeUpdate(
+          // language=sql
+          """
+          INSERT INTO
+            merlin.external_event
+          VALUES ('%s', '%s', '%s', '%s', '%s', '%s');
+          """.formatted(
+              externalEvent.key,
+              externalEvent.event_type_name,
+              externalEvent.source_key,
+              externalEvent.derivation_group_name,
+              externalEvent.start_time,
+              externalEvent.duration
+          )
+      );
+    }
+  }
+
+  protected void associateDerivationGroupWithPlan(int planId, String derivationGroupName) throws SQLException {
+    try(final var statement = connection.createStatement()) {
+      // create the event type
+      statement.executeUpdate(
+          // language=sql
+          """
+          INSERT INTO
+            merlin.plan_derivation_group
+          VALUES ('%s', '%s');
+          """.formatted(planId, derivationGroupName)
+      );
+    }
+  }
+
+  /**
+   * Quick external event creator, leveraging constants from a provided source object (source).
+   */
+  protected ExternalEvent createEvent(String key, String start_time, String duration, ExternalSource source) {
```

Comment:
This can be a constructor on the `ExternalEvent` record:

```java
  protected record ExternalEvent(
      String key,
      String event_type_name,
      String source_key,
      String derivation_group_name,
      String start_time,
      String duration
  ) {
    ExternalEvent(String key, String start_time, String duration, ExternalSource source) {
      this(key, et, source.key(), source.derivation_group_name(), start_time, duration);
    }
  }
```

Doing this would require `et` to become `static`, but to be honest all of the variables in `Commonly Repeated Variables` can become `static`, as they're constants that will have the same value regardless of class instance.